types/field_type.go

```diff
@@ -155,6 +155,8 @@ func (ft *FieldType) CompactStr() string {
 	case mysql.TypeBit, mysql.TypeShort, mysql.TypeTiny, mysql.TypeInt24, mysql.TypeLong, mysql.TypeLonglong, mysql.TypeVarchar, mysql.TypeString, mysql.TypeVarString:
 		// Flen is always shown.
 		suffix = fmt.Sprintf("(%d)", displayFlen)
+	case mysql.TypeYear:
+		suffix = "(4)"
```

Comment:
Please add a test for this.

Reply:
@spongedu Please address this comment.

Reply:
@zz-jason I've migrated `field_type_test.go` from TiDB repo to parser. See: https://github.com/pingcap/parser/pull/200 . I think we can add a test for this in `field_type_test.go` after #200  is merged. What do you think ?

Reply:
OK, let's work on https://github.com/pingcap/parser/pull/200 firstly.

Reply:
@spongedu How about only adding errors in this PR? So that this PR and https://github.com/pingcap/tidb/pull/9239 can be merged without https://github.com/pingcap/parser/pull/200

Reply:
@zz-jason OK, I'll remove this change for now